### PR TITLE
PitchYinFFT updates (new weighting parameter, do not throw exception on empty peak detection)

### PR DESCRIPTION
**Pull request by xaviliz**
1) Add weighting parameter to select different weighting funcitons in PITCHYINFFT algorithm. The available choices are: 
    - the default weighting function
    - A-weighting
    - B-weighting
    - C-weighting
    - D-weighting
    - Z-weighting.

    Note that there is not experimental validation of better pitch estimation performance with new alternative weightings. 

2) Change behavior on empty spectrum peak detection: output zero pitch and confidence instead of throwing an exception.

@dbogdanov Edited for completeness.

### CODE REVIEW

File: src/algorithms/tonal/pitchyinfft.cpp [hunk 1]
```diff
@@ -189,8 +245,9 @@ void PitchYinFFT::compute() {
       yinMin = -_amplitudes[0];
     }
     else {
-      // TODO this should never happen, but some people reported it happening in their real time applications.
-      throw EssentiaException("PitchYinFFT: it appears that no peaks were found by PeakDetection. If you read this message, PLEASE, report this issue to the developers with an example of audio on which it happened.");
+      tau = 0.0; // it will provide zero-pitch and zero-pitch confidence.
```
Comment (dbogdanov) [inline src/algorithms/tonal/pitchyinfft.cpp:248]: This change is out of scope of weighting functions.  At least would should add the description of this change to the PR.

Comment (xaviliz) [reply src/algorithms/tonal/pitchyinfft.cpp:248]: Oh yes! this is something we need in the ODISEI project to avoid a crucial error. Indeed we would need to launch a negative pitch value. No problem I will redo what essentia does in this cases.

Comment (dbogdanov) [reply src/algorithms/tonal/pitchyinfft.cpp:248]: This change had been already included in the merge.
